### PR TITLE
Add Integration test for `ExpandVolume`

### LINKED ISSUE #340: ExpandVolume Method Testing
# Summary
Test ExpandVolume Method

## Description
- Create test cases to validate the functionality of the `ExpandVolume` method.
- Use the `go-ceph` library to confirm that volumes can be expanded as expected in the Ceph cluster.
- Test edge cases where expansion might fail or succeed.

### PR DESCRIPTION
**Pull request by ushabelgur**
# Proposed Changes

- Integration test for expand volume

Fixes #340

### CODE REVIEW

**Review by afritzler — CHANGES_REQUESTED** [review]

File: integration/volume_expand_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+			HaveField("Volume.Metadata.Id", Not(BeEmpty())),
+			HaveField("Volume.Spec.Image", Equal("")),
+			HaveField("Volume.Spec.Class", Equal("foo")),
+			HaveField("Volume.Spec.Resources.StorageBytes", Equal(uint64(1024*1024*1024))),
+			HaveField("Volume.Spec.Encryption", BeNil()),
+			HaveField("Volume.Status.State", Equal(onmetalv1alpha1.VolumeState_VOLUME_PENDING)),
+			HaveField("Volume.Status.Access", BeNil()),
+		))
+
+		// Ensure the correct image has been created inside the ceph cluster
```
Comment (afritzler) [inline integration/volume_expand_integration_test.go:61]: Can we remove the comments and instead add a `By` statement containing the description on what is going to happen now?

**Review by afritzler — CHANGES_REQUESTED** [review]

File: integration/volume_expand_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+					Id: "foo",
+				},
+				Spec: &onmetalv1alpha1.VolumeSpec{
+					Class: "foo",
+					Resources: &onmetalv1alpha1.VolumeResources{
+						StorageBytes: 1024 * 1024 * 1024,
+					},
+				},
+			},
+			XXX_sizecache: 1024,
```
[the remark this answers is not in the document]
Comment (ushabelgur) [reply integration/volume_expand_integration_test.go]: removed

File: integration/volume_expand_integration_test.go (hunk 1 above)
Comment (ushabelgur) [reply integration/volume_expand_integration_test.go]: Done

**Review by afritzler — APPROVED** [review]

**Review by lukasfrank — APPROVED** [review]
